plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -272,7 +297,13 @@ public IcebergTableHandle getTableHandle(ConnectorSession session, SchemaTableNa
         catch (TableNotFoundException e) {
             return null;
         }
-        Optional<Long> snapshotId = getSnapshotId(table, name.getSnapshotId());
+
+        if (name.getSnapshotId().isPresent() && endVersion.isPresent()) {
+            throw new TrinoException(GENERIC_USER_ERROR, "Cannot specify end version both in table name and FOR clause");
```

Comment:
`FOR` -> `AS OF`?

I guess this also means we could deprecate the `table$version` method, @findepi

Reply:
> guess this also means we could deprecate the `table$version` method, @findepi

yes, see https://github.com/trinodb/trino/pull/10768#issue-1112989759